### PR TITLE
Add new auto case creation fields to the transformation end point

### PR DESCRIPTION
**Pull request by aliveni**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/BPS-1304

### Change description ###
Update transformation request to include new fields required for auto case creation.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Comment by aliveni** [comment]
> Worth writing a `TransformationController` test which would make sure if you pass old and new fields both works fine ?

I haven't made any functionality changes yet. for now, it just works with the existing endpoint.

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java [hunk 1]
```diff
@@ -29,7 +33,11 @@ public ExceptionRecord(
         @JsonProperty("delivery_date") LocalDateTime deliveryDate,
         @JsonProperty("opening_date") LocalDateTime openingDate,
         @JsonProperty("scanned_documents") List<InputScannedDoc> scannedDocuments,
-        @JsonProperty("ocr_data_fields") List<OcrDataField> ocrDataFields
+        @JsonProperty("ocr_data_fields") List<OcrDataField> ocrDataFields,
+        @JsonProperty("envelope_id") String envelopeId,
+        @JsonProperty("is_automated_process") String isAutomatedProcess,
+        @JsonProperty("exception_record_id") String exceptionRecordId,
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java:39]: is that not the `id` property on top?

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java [hunk 2]
```diff
@@ -29,7 +33,11 @@ public ExceptionRecord(
         @JsonProperty("delivery_date") LocalDateTime deliveryDate,
         @JsonProperty("opening_date") LocalDateTime openingDate,
         @JsonProperty("scanned_documents") List<InputScannedDoc> scannedDocuments,
-        @JsonProperty("ocr_data_fields") List<OcrDataField> ocrDataFields
+        @JsonProperty("ocr_data_fields") List<OcrDataField> ocrDataFields,
+        @JsonProperty("envelope_id") String envelopeId,
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java:37]: i think i can re-use that while refactoring orchestrator. but that's for future

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java [hunk 3]
```diff
@@ -29,7 +33,11 @@ public ExceptionRecord(
...
+        @JsonProperty("is_automated_process") String isAutomatedProcess,
```
Comment (nitinprabhu) [inline src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java:38]: may be we can name class field  differently automatedProcessFlag as it is not boolean ?

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java (hunk 1 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java:39]: yeah, it is `id` but it should be renamed as `exception_record_id` for the new endpoint. 
same for `case_type_id`. `case_type_id` will be renamed a s`exception_record_case_type_id`.
We have to keep all the above fields to support both old and new transformation endpoints

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java (hunk 3 above)
Comment (aliveni) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java]: I changed the field type as `boolean`

File: src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java (hunk 1 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/bulkscanccdeventhandler/common/model/in/ExceptionRecord.java:39]: ah rename process. ok, got you

**Pull request merged**